components/src/core/components/Input/PasswordInput.vue

```diff
@@ -1,16 +1,141 @@
 <template>
-  <oxd-input type="password" />
+  <div :class="strengthClass">
+    <oxd-text class="password-strength-check" v-if="showStrength">{{
+      strengthLabel
+    }}</oxd-text>
+    <oxd-input
+      :type="isPasswordVisible ? 'text' : 'password'"
+      class="password-input"
+      v-bind="$attrs"
+      :hasError="hasError"
+      :disabled="disabled"
+      :readonly="readonly"
+    />
+    <div
+      :class="[
+        'password-view-icon-container',
+        {'password-visible': isPasswordVisible},
+      ]"
+      @click="togglePasswordVisibility"
+    >
+      <oxd-icon-button
+        :name="isPasswordVisible ? 'eye-slash-fill' : 'eye-fill'"
+        class="oxd-password-view-icon"
+        :iconStyles="isPasswordVisible ? {color: 'white'} : {}"
+        size="large"
+        :tooltip="isPasswordVisible ? 'Hide Password' : 'Show Password'"
+      />
+    </div>
+  </div>
 </template>
 
 <script lang="ts">
 import {defineComponent} from 'vue';
 import Input from '@orangehrm/oxd/core/components/Input/Input.vue';
+import oxdText from '@orangehrm/oxd/core/components/Text/Text.vue';
+import IconButton from '@orangehrm/oxd/core/components/Button/Icon.vue';
+
+export interface State {
+  showStrength: boolean;
+  isPasswordVisible: boolean;
+}
 
 export default defineComponent({
   name: 'oxd-password-input',
 
   components: {
     'oxd-input': Input,
+    'oxd-text': oxdText,
+    'oxd-icon-button': IconButton,
+  },
+  props: {
+    strength: {
+      type: Number,
+      default: -1,
```

Comment:
strength will need a validator to prevent arbitrary strength values.
https://vuejs.org/guide/components/props.html#prop-validation

Reply:
strength: {
      type: Number,
      default: -1,
      validator: (value: number) => value >= -1 && value <= 5,
    },